### PR TITLE
Add HttpPollWorkerTests e2e tests for http poll task

### PR DESCRIPTION
**Pull request by supriyanta**

### CODE REVIEW

File: src/test/java/io/orkes/conductor/client/e2e/HttpPollWorkerTests.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        try {
+            metadataClient.unregisterWorkflowDef(WORKFLOW_NAME, 1);
+            workflowClient.deleteWorkflow(WORKFLOW_NAME, false);
+        } catch (Exception e) {
+        }
+    }
+
+    @Test
+    public void testHttpPollTask() {
+        String url = "https://jsonplaceholder.typicode.com/posts";
```
Comment (v1r3n) [inline src/test/java/io/orkes/conductor/client/e2e/HttpPollWorkerTests.java:57]: don't use external service.  Use orkesapi endpoint we have.  Check playground -- when you create a new http task, it automatically adds a test url - we can use that.

Comment (supriyanta) [reply src/test/java/io/orkes/conductor/client/e2e/HttpPollWorkerTests.java:57]: Sure, let me change that to it

**Comment by manan164** [comment]
There few test cases I can think of,
1. Test fixed interval http poll
2. Test Linear back off http poll
3. Test exponential back off http poll.
In above we can use interval as low as 3 seconds so that overall test wont exceed 10-20 seconds.
4. Test error in evaluating terminal condition.
5. Test terminal condition with out come as 0,1,-1. We do support that. This can be three separate test.

**Comment by supriyanta** [comment]
@manan164 Interval can't be less than 60 secs

**Comment by manan164** [comment]
Hi @supriyanta , As discuseed over IM,  We can assert on task' callBackAfterSeconds in that case. We just need to verify that the backoff works. We really need not to wait till the time.

**Comment by supriyanta** [comment]
@manan164  Added all the cases. Please review the changes

**Pull request closed without merging**